### PR TITLE
Configure /api proxy with SSL options

### LINKED ISSUE #166: /api/* endpoint does not configure SSL properly
We're configuring HTTPS for the express server properly. The /api proxy endpoint relies on a createProxyServer call which must also accept an `ssl` parameter bundle with the same key/cert as the main server.

### PR DESCRIPTION
**Pull request by parente**
Fixes #166

### CODE REVIEW

**Comment by parente** [comment]
@dalogsdon please give this a look see

File: app/config.js [hunk 1]
```diff
@@ -26,4 +27,20 @@ config.set('NOTEBOOKS_DIR', path.join(__dirname, '..', config.get('NOTEBOOKS_DIR
...
+    }
+    if(!fs.existsSync(cert_file_location)) {
+        throw new Error('Invalid file path for HTTPS_CERT_FILE');
+    }
+    config.set('SSL_OPTIONS', {
+       key: fs.readFileSync(key_file_location),
+       cert: fs.readFileSync(cert_file_location)
+    });
+}
+
```
Comment (dalogsdon) [inline app/config.js]: :+1: for moving config validation to config.js

**Comment by dalogsdon** [comment]
The change works for me, but I also see things work without the change. Looked into it with @parente, and we see in `bin/www` that it looks like the https server will handle this for us by passing in the `app` file.

**Comment by parente** [comment]
Will update to keep the code change that moves the logic to config out of `www`, but will not inject the ssl options into the proxy server.

I'm miffed about why I was seeing `POST /api/kernels?1459777914810 - - ms - -` earlier. Could be same rare problem as #156 ?

**Comment by parente** [comment]
@dalogsdon Kept the config cleanup but removed use of the SSL options on the `createProxyServer` call. Added a comment about why it's not config'ed there.

**Pull request merged**